src/ray/gcs/gcs_server/gcs_node_manager.h

```diff
@@ -7,7 +7,89 @@
 
 namespace ray {
 
+namespace rpc {
+class NodeManagerWorkerClient;
+}
+
 namespace gcs {
+class GcsActor;
+class GcsLeasedWorker;
+/// The GcsNode is kind of abstraction which corresponds to the real raylet one by one.
+/// It provides interface to lease workers from the real raylet via
+/// `NodeManagerWorkerClient` and manage them.
+/// Also, it is easily to get the NodeID and Endpoint of the ralated raylet from the gcs
+/// node.
+class GcsNode : public std::enable_shared_from_this<GcsNode> {
```

Comment:
This class does not seem necessary. It appears to be a thin wrapper around a hashmap. I do not see any reason why the `GcsNodeManager` and the `GcsActorManager` need to share state through this class. The split of functionality should be:

`GcsNodeManager` -> Maintains which node IDs are alive and dead. Calls directly into `GcsActorManager` when a node is added/removed. Does not need to know anything about leased workers.
`GcsActorManager` -> Maintains map from node ID to leased workers.

The only common state should be about node IDs.